src/cms/views/push_notifications/push_notification_sender.py

```diff
@@ -1,43 +1,133 @@
+"""
+Module for sending Push Notifications
+"""
 import logging
 import requests
 
-from ...models import Configuration
+from django.conf import settings
 
+from ...models import Configuration
+from ...models import PushNotificationTranslation
+from ...constants import push_notifications as pnt_const
 
 # pylint: disable=too-few-public-methods
 class PushNotificationSender:
+    """
+    Sends push notifications via FCM HTTP API.
+    Definition: https://firebase.google.com/docs/cloud-messaging/http-server-ref#downstream-http-messages-json
+    """
+
     logger = logging.getLogger(__name__)
     fcm_url = "https://fcm.googleapis.com/fcm/send"
 
-    """
-    Sends push notifications via FCM legacy http api.
-    Returns boolean indicating success or failure
-    """
+    def __init__(self, push_notification):
+        """
+        Load relevant push notification translations and prepare content for sending
 
-    # pylint: disable=too-many-arguments
-    def send(self, region_slug, channel, title, message, lan_code):
+        :param push_notification: the push notification that should be sent
+        :type push_notification: ~cms.models.push_notifications.push_notification.PushNotification
+        """
+        self.push_notification = push_notification
+        self.prepared_pnts = []
+        self.primary_pnt = PushNotificationTranslation.objects.get(
+            push_notification=push_notification,
+            language=push_notification.region.default_language,
+        )
+        if len(self.primary_pnt.title) > 0:
+            self.prepared_pnts.append(self.primary_pnt)
+        self.load_secondary_pnts()
+        self.auth_key = self.get_auth_key()
+
+    def load_secondary_pnts(self):
+        """
+        Load push notification translations in other languages
+        """
+        secondary_pnts = PushNotificationTranslation.objects.filter(
+            push_notification=self.push_notification
+        ).exclude(id=self.primary_pnt.id)
```

Comment:
just a side note for the future: you can also use the reverse related name in such cases: `self.push_notification.translations.exclude(id=self.primary_pnt.id)`